### ISSUE TITLE
Shader function not debuggable?

### ISSUE THREAD

**Issue by TomasBlaho** [baldurk/renderdoc#969]
Hello,
I'm debugging a pixel shader and I can see the usual mixture of HLSL lines and asm as I'm compiling shaders in Unity 2017.3 using #pragma enable_d3d11_debug_symbols. The problem is if I call a function, there is just an assembly for it and no HLSL. Could it be an issue of RenderDoc? If I insert just the code of the function instead of calling it, it's debuggable. I tried both RenderDoc 1.0 stable and the latest nightly build (25-Apr-2018).
Regards,
Tomas

**Comment by baldurk** [comment]
Can you give more details, e.g. are you able to share a capture that exhibits the problem? It's not clear from your description exactly what you're changing and what is going wrong.

It could be a bug with renderdoc not being able to fetch the source correctly from the embedded debug info, or it could be something else entirely. I'll need to be able to reproduce the problem to investigate it.

**Comment by TomasBlaho** [comment]
Hello baldurk,
I recently changed me work email, so I didn't notice the notification about your replay. Now it should be fixed in my account. 
Regarding the issue, I made a capture for you [HERE](https://drive.google.com/file/d/1Bxow8uNhbxs2xYjOpdkaKCk5vD9gMhLc/view?usp=sharing)
Event 10680 shows a drawcall with debug info for PS enabled, but a function TLCLightPixel included from tiledLightsculling.cginc, has no debug info and is shown as a block of assembly. I have to embed the code of the function to the main file to have debug info for it (event 11625). It's probably a Unity thing. Looks like they pre-compile .cginc header with no debug info maybe? I'm sorry I'm wasting your time if that's the case.
Regards,
Tomas

**Comment by baldurk** [comment]
It's not possible to precompile cginc headers, fxc compiles everything all at once and generates combined debug info. It would all be generated or none of it.

From looking at the capture it seems like renderdoc isn't probably parsing the embedded pdb line number data. It was originally reverse engineered so hopefully it shouldn't be too hard to fix - there might be something useful in the pdb docs microsoft published a while ago too.

**Comment by TomasBlaho** [comment]
I'm currently embedding the code I want to debug everywhere, but if you can fix that issue, it's going to be much easier for me. Thanks!

**Comment by baldurk** [comment]
I forgot to comment here - I've fixed the underlying issue here in that commit. I went back to the released PDB spec and with some other external documentation to fill in the missing gap I was able to replace most of my reverse engineered code. I got pretty close, but there were a couple of bugs :smile:.

That should now correctly have all the line numbers for called functions, but fully decoding all the PDB data has unlocked some other information that I'm working on plumbing through to the UI. Watch this space...
